tests/bdd/features/volume/resize/test_resize_offline.py

```diff
@@ -0,0 +1,448 @@
+        if err.status == 404:
+            ApiClient.pools_api().put_node_pool(
+                NODE4_NAME, POOL4_UUID, CreatePoolBody([f"aio://{disks[3]}"])
+            )
+    yield volume
+    Volume.cleanup(volume)
+
+
+@given("a published volume with more than one replica and all are healthy")
+def a_published_volume_with_more_than_one_replica_and_all_are_healthy(create_volume):
+    """a published volume with more than one replica and all are healthy."""
+    the_volume_is_published()
+    pytest.exception = None
+
+
+@given(
+    parsers.parse(
+        "an unpublished volume with {repl_count:d} replicas and all are healthy"
+    )
+)
+@given("an unpublished volume with <repl_count> replicas and all are healthy")
+def an_unpublished_volume_with_repl_count_replicas_and_all_are_healthy(repl_count):
+    """an unpublished volume with <repl_count> replicas and all are healthy."""
+    pytest.exception = None
+    pytest.replica_count = repl_count
+    volume = ApiClient.volumes_api().put_volume(
+        VOLUME_UUID,
+        CreateVolumeBody(VolumePolicy(True), int(repl_count), VOLUME_SIZE, False),
+    )
+    assert volume.spec.uuid == VOLUME_UUID
+    replicas = volume.state.replica_topology
+    assert len(replicas) == int(repl_count)
+    yield volume
+    Volume.cleanup(volume)
+
+
+@given("an unpublished volume with more than one replica")
+def an_unpublished_volume_with_more_than_one_replica(create_volume):
+    """an unpublished volume with more than one replica."""
+
+
+@given("an unpublished volume with more than one replica and all are healthy")
+def an_unpublished_volume_with_more_than_one_replica_and_all_are_healthy(create_volume):
+    """an unpublished volume with more than one replica and all are healthy."""
+
+
+@when("one of the replica goes offline")
+def one_of_the_replica_goes_offline():
+    """one of the replica goes offline."""
+    # Offline NODE3_NAME to make the replica Offline/Unknown
+    Docker.stop_container(NODE3_NAME)
+    wait_child_faulted()
+
+
+@given("one of the replica is not in online state")
+def one_of_the_replica_is_not_in_online_state():
+    """one of the replica is not in online state."""
+    pytest.exception = None
+    volume = ApiClient.volumes_api().get_volume(VOLUME_UUID)
+    # Offline NODE3_NAME to make the replica Offline/Unknown
+    Docker.stop_container(NODE3_NAME)
+    yield
+    the_replica_comes_online_again()
+
+
+@when("the volume is receiving IO")
+def the_volume_is_receiving_io():
+    """the volume is receiving IO."""
+    volume = ApiClient.volumes_api().get_volume(VOLUME_UUID)
+    uri = urlparse(volume.state.target["deviceUri"])
+    fio = Fio(name="fio-pre-resize", rw="write", uri=uri, extra_args="--loops=2")
+    pytest.fio = fio.open()
+
+
+@when("the volume is unpublished")
+def the_volume_is_unpublished():
+    """the volume is unpublished."""
+    # Ensure fio is still running till unpublish
+    time.sleep(1)
+    assert pytest.fio.poll() is None
```

Comment:
```suggestion
try:
    code = pytest.fio.wait(timeout=2)
    assert code == 0, "FIO failed, exit code: %d" % code
except subprocess.TimeoutExpired:
    assert False, "FIO timed out"
```

Reply:
`wait` will block till fio completion here for a timeout of 2sec. However, with `poll` I'm just making sure the fio process is still running in background when unpublish is done. `sleep` is optional though.

Reply:
hmm but how can you unpublish the volume if fio is still running?